packages/neos-ui/src/Containers/Modals/SelectNodeType/index.js

```diff
@@ -60,7 +62,9 @@ export default class SelectNodeType extends PureComponent {
     };
 
     componentDidMount() {
+        const {filterSearchTerm} = this.state || {};
         this.setState({
+            filterSearchTerm: filterSearchTerm || '',
```

Comment:
Wait, but why is it even needed? Can you explain?

Reply:
Because we use it later on. On initial load it might be null wich is a problem with the regex check etc wich needs a string. So we keep here the value in filterSearchTerm or set it to an empty string. The keep thing will kick in on every `componentDidUpdate` hook, the empty string when   `const {filterSearchTerm} = this.state || {};` is falsy

Reply:
The main problem here is that `componentDidUpadte` triggers `componentDidMount` and we need to keep the state as well as initialize the state